src/pages/BalanceHistory/partials/BalanceBodyList.tsx

```diff
@@ -0,0 +1,90 @@
+import { cn } from "@utils/cn";
+import { useEffect } from "react";
+import { useBalanceHistoryContext } from "../context";
+import AppearGrow from "@components/Animation/AppearGrow";
+import AppearFadeIn from "@components/Animation/AppearFadeIn";
+import glassmorphism from "@utils/glassmorphism";
+
+type BalanceBodyListProps = {
+  firstNewDate: boolean;
+  setFirstNewDate: (value: boolean) => void;
+};
+
+const BalanceBodyList = ({
+  firstNewDate,
+  setFirstNewDate,
+}: BalanceBodyListProps) => {
+  useEffect(() => {
+    if (firstNewDate) {
+      setFirstNewDate(false);
+    }
+  }, [firstNewDate, setFirstNewDate]);
+  const { state } = useBalanceHistoryContext();
+  let lastDate = "";
+
+  return (
+    <AppearGrow trigger direction="x">
+      {state.balanceHistory.map((item, index) => {
+        const currentDate = item.created_at.split("at")[0];
+        const isNewDate = currentDate !== lastDate;
+        lastDate = currentDate;
+
+        return (
+          <AppearFadeIn key={index} direction="bottom" delay={0.1 * index}>
+            {isNewDate && (
+              <div
+                className={cn(
+                  "p-4 font-semibold",
+                  firstNewDate && "rounded-t-md",
+                )}
+              >
+                {currentDate}
+              </div>
+            )}
+            <div
+              className={cn(
+                "p-4 flex justify-between group",
+                glassmorphism({ container: true, hover: true }),
+              )}
+            >
+              <div className="flex flex-col gap-1 font-medium group-hover:text-gray-300">
+                <div>
```

Comment:
unecessary div, jangan kebanyakan beranak

Reply:
there are no unnecessary div, duyu think unclassnamed div are doesnt needed ? no, that div role are to manage flex layout and right now there are no styling applied to that div

Reply:
berarti pake span, kan cuma text